solr/packaging/build.gradle

```diff
@@ -62,12 +63,17 @@ dependencies {
 
   example project(path: ":solr:example", configuration: "packaging")
   server project(path: ":solr:server", configuration: "packaging")
+
+  // Copy files from documentation output
+  docs project(path: ':solr', configuration: 'docs')
```

Comment:
I have no idea where this configuration is defined. The `:solr` project only has a task called `documentation` that needs to be executed before, but the outputs are not defined as config. So how should this work (same for Lucene)?

Reply:
There are some `ext` properties prointing to the documentation after it's generated. For the Solr ZIP/TGZ its just some minimal folder: `project(':solr').docrootOnline` (packaged in ZIP)

For full lucene/solr docs, refer to `project(':lucene').docroot` (packaged in TGZ) and `project(':solr').docroot` (not packaged, only published on website).

Reply:
I explained this above, I hope. I think artifact/configuration based approach to collecting packaging dependencies is superior to explicit properties and task-dependencies.

Reply:
But where is the documentation config defined. I did not find it anywhere else. I understand and I like it, but this is incomplete. And currently it does not work (I see no docs in ZIP files).

Reply:
So as far as I understand: In the documentation.gradle file I would add a "configuration docs" for `:solr` and `:lucene` and then use `builtBy 'documentation'` there. That's what I am missing here. I was about to set this up yesterday, but was not sure if I miss something.

Reply:
Ah, sorry - yes, you are correct. It should declare artifacts, the configuration to attach them to and the outputs. An example is in solr/example/build.gradle:

artifacts {
  packaging packagingDir, {
    builtBy assemblePackaging
  }
}

Reply:
Ah cool. That is what I was missing. I will add this after lunch. I was not sure how to define the artifacts. The Gradle documentation was not very helpful to me.
Thanks and sorry for stupid questions. 🤣

Reply:
There are no stupid questions. This is the relevant documentation bit here, I believe.
https://docs.gradle.org/current/userguide/cross_project_publications.html